src/pymor/operators/block.py

```diff
@@ -203,10 +227,21 @@ def assemble(self, mu=None):
             return self.__class__(blocks)
 
     def assemble_lincomb(self, operators, coefficients, solver_options=None, name=None):
+        if not all(isinstance(op, BlockOperator) for op in operators):
+            return None
+
+        assert operators[0] is self
+
+        # return ShiftedSecondOrderSystemOperator if possible
+        if len(operators) == 2 and isinstance(operators[1], SecondOrderSystemOperator):
+            return operators[1].assemble_lincomb(operators[::-1], coefficients[::-1],
+                                                 solver_options=solver_options, name=name)
```

Comment:
Yeah, it really isn't beatiful to have this special case here. I guess this shows that `assemble_lincomb` has its limitations. But I do not think that we should try to do something about this befor the 0.5 release. I will open a new issue ...